### DISCUSSION TITLE
Docker Error: Killed when Deploying Function

### DISCUSSION THREAD

**Discussion by tanmoy27112000** [appwrite/appwrite#4839] [Q&A] [answered]
### 👟 Reproduction steps

Create a sample dart function and copy the dart code from the docs 

```dart
import 'dart:math';
import 'dart:async';

Future<void> start(final req, final res) async {
  final payload = !req.payload?.isEmpty
      ? req.payload
      : 'No payload provided. Add custom data when executing function.';

  final secretKey = req.variables['SECRET_KEY'] ??
      'SECRET_KEY variable not found. You can set it in Function settings.';

  final randomNumber = Random().nextDouble();

  final trigger = req.variables['APPWRITE_FUNCTION_TRIGGER'];

  res.json({
    'message': 'Hello from Appwrite!',
    'payload': payload,
    'secretKey': secretKey,
    'randomNumber': randomNumber,
    'trigger': trigger,
  });
}
```

deploy it.

### 👍 Expected behavior

The deployment should build successfully.

### 👎 Actual Behavior

getting error docker killed.
![Screenshot 2022-12-09 at 2 13 42 AM](https://user-images.githubusercontent.com/22212661/206563758-bbc05f64-1253-4c67-94ec-001098597187.png)

### 🎲 Appwrite version

Version 1.1.x

### 💻 Operating system

MacOS

### 🧱 Your Environment

I am using the latest one.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
The most common reason why `Docker Error: Killed` occurs is because the containers runs out of memory. Please try increasing the following [environment variables](https://appwrite.io/docs/environment-variables#functions):

* `_APP_FUNCTIONS_MEMORY`
* `_APP_FUNCTIONS_MEMORY_SWAP`

For reference, I have mine set to 2048.
